libelement/src/token_internal.hpp

```diff
@@ -37,8 +37,8 @@ struct element_tokeniser_ctx
     element_token* get_token(std::size_t token_index, element_result& out_result);
 
     std::shared_ptr<element_log_ctx> logger = nullptr;
-    const char* raw_filename = nullptr;
-    std::string filename;
+    const char* raw_source_name = nullptr;
+    std::string source_name;
```

Comment:
can you look at removing source_name entirely? I don't think it's something we want to use, the `raw_source_name` pointer is important to preserve right now.